GREEDY_WEB/ClientApp/components/UserLogin.tsx

```diff
@@ -0,0 +1,116 @@
+﻿import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+import { Button } from 'reactstrap';
+import DocumentTitle from 'react-document-title';
+import { LoginForm } from 'react-stormpath';
+import axios from 'axios';
+import { Link, NavLink } from 'react-router-dom';
+import Cookies from 'universal-cookie';
+import FacebookLogin from 'react-facebook-login';
+
+export class UserLogin extends React.Component<RouteComponentProps<{}>> {
+    constructor(props) {
+        super(props);
+        this.onFormSubmit = this.onFormSubmit.bind(this);
+    }
+
+    onFormSubmit(e, next) {
+        e.preventDefault();
+        var data = e.data;
+
+        if (data.username.length < 1) {
+            return next(new Error('Username/Email must not be empty.'));
+        }
+
+        if (data.username.length > 256) {
+            return next(new Error('Username/Email is too long.'));
+        }
+
+        // Require passwords to be at least 10 characters.
+        if (data.password.length < 1) {
```

Comment:
same as in username. we have to have 1 as a constant and for security purpose we need a longer password too, Also, we need to add more if for upper and lower case and etc. I comment to have additional method for password verification.